quic/s2n-quic-qns/src/tls.rs

```diff
@@ -212,6 +225,41 @@ pub mod s2n_tls {
             s2n_quic_core::crypto::tls::testing::certificates::KEY_PEM.into_private_key()?
         })
     }
+
+    #[derive(Clone)]
+    pub struct SessionTicketHandler {
+        ticket_storage: Arc<Mutex<VecDeque<Vec<u8>>>>,
+    }
+
+    impl Default for SessionTicketHandler {
+        fn default() -> SessionTicketHandler {
+            SessionTicketHandler {
+                // Cap session ticket storage at 10 tickets
+                ticket_storage: Arc::new(Mutex::new(VecDeque::with_capacity(10))),
+            }
+        }
+    }
+
+    impl SessionTicketCallback for SessionTicketHandler {
+        fn on_session_ticket(&self, _connection: &mut Connection, session_ticket: &SessionTicket) {
+            let size = session_ticket.len().unwrap();
+            let mut data = vec![0; size];
+            session_ticket.data(&mut data).unwrap();
+            let mut vec = (*self.ticket_storage).lock().unwrap();
+            vec.push_back(data);
```

Comment:
this is how you'd limit the queue size

```suggestion
            let mut vec = (*self.ticket_storage).lock().unwrap();
            vec.push_back(data);
            // discard any excessive tickets
            while vec.len() > 10 {
                let _ = vec.pop_front();
            }
```